scheduler/src/test/java/io/camunda/zeebe/scheduler/lifecycle/ActorLifecyclePhasesTest.java

```diff
@@ -351,4 +352,137 @@ public void onActorStarted() {
                 ActorLifecyclePhase.STARTED,
                 ActorLifecyclePhase.FAILED));
   }
+
+  @Test
+  public void shouldCompleteCloseFutureWhenFailingInStarted() {
+    // given
+    final LifecycleRecordingActor actor =
+        new LifecycleRecordingActor() {
+          @Override
+          public void onActorStarted() {
+            super.onActorStarted();
+            actor.fail(new RuntimeException("foo"));
+          }
+        };
+
+    // when - explicitly do not work after close as it should be completed already
+    schedulerRule.submitActor(actor);
+    schedulerRule.workUntilDone();
+    final ActorFuture<Void> closeFuture = actor.closeAsync();
+
+    // then
+    assertThat(actor.phases)
+        .isEqualTo(
+            List.of(
+                ActorLifecyclePhase.STARTING,
+                ActorLifecyclePhase.STARTED,
+                ActorLifecyclePhase.FAILED));
+    assertThat(closeFuture).succeedsWithin(Duration.ofSeconds(1));
+  }
+
+  @Test
+  public void shouldCompleteCloseFutureOnExceptionOnClosing() {
+    // given
+    final LifecycleRecordingActor actor =
+        new LifecycleRecordingActor() {
+          @Override
+          public void onActorClosing() {
+            super.onActorClosing();
+            throw new RuntimeException("foo");
+          }
+        };
+
+    // when
+    schedulerRule.submitActor(actor);
+    schedulerRule.workUntilDone();
+    final ActorFuture<Void> closeFuture = actor.closeAsync();
+    schedulerRule.workUntilDone();
+
+    // then
+    assertThat(actor.phases)
+        .isEqualTo(
+            List.of(
+                ActorLifecyclePhase.STARTING,
+                ActorLifecyclePhase.STARTED,
+                ActorLifecyclePhase.CLOSE_REQUESTED,
+                ActorLifecyclePhase.CLOSING));
+    assertThat(closeFuture).failsWithin(Duration.ofSeconds(1));
+  }
+
+  @Test
+  public void shouldCompleteCloseFutureWhenFailingOnClosing() {
```

Comment:
:x: For the sake of completion, we should also add test for "close_requested" phase. `shouldCompleteCloseFutureWhenFailingOnCloseRequested` and `shouldCompleteCloseFutureWhenExceptionOnCloseRequested`

Reply:
Interestingly, if you throw an exception in close requested, we will keep going to `closing` then `closed`. OTOH this seems acceptable, though maybe a little strange that here it doesn't short circuit the lifecycle (e.g. throwing an exception in closing means you stop there).

Reply:
hmm.. interesting. I thought it will go to FAIL.

Reply:
If you `fail` then it does. But if you just throw an exception it keeps going. It might be what I changed here, i.e. that we only handle failures/go to fail in STARTED when an exception is thrown. I can add CLOSE_REQUESTED as well, if that's the expected behavior.

Reply:
Actually, come to think of it, should we not treat CLOSE_REQUESTED and CLOSING the same in this instance?

Reply:
Yes. I think so. In `ActorTask::onFailure` we are not currently handling it CLOSE_REQUESTED. But it should be.